### PR TITLE
CURA-11138-makerbot-cloud-printing

### PR DESCRIPTION
**Pull request by saumyaj3**
CURA-11138

Makerbot printers in cloud

### CODE REVIEW

**Review by rburema — CHANGES_REQUESTED** [review]
Nice work!

I have a couple of small remarks.

I'm also not sure about the amount of hardcoding -- but I guess it can't be helped unless we want to spend a bunch more time on this (and also, we already do some similar things for UM printers...)

In the mean-time, I'll move this to component-test, so I can also check if all the subtasks have been done.

File: cura/PrinterOutput/Models/ExtruderConfigurationModel.py [hunk 1]
```diff
@@ -40,9 +40,22 @@ def material(self) -> Optional[MaterialOutputModel]:
 
     def setHotendID(self, hotend_id: Optional[str]) -> None:
         if self._hotend_id != hotend_id:
-            self._hotend_id = hotend_id
+            self._hotend_id = ExtruderConfigurationModel.getHotendIdIfMakerbot(hotend_id)
             self.extruderConfigurationChanged.emit()
 
+    @staticmethod
+    def getHotendIdIfMakerbot(hotendId) -> str:
```
[the remark this answers is not in the document]
Comment (saumyaj3) [reply cura/PrinterOutput/Models/ExtruderConfigurationModel.py]: ce132e4c3f5292358a15c78ccd29aaace3e31f06

File: plugins/DigitalLibrary/src/DFFileExportAndUploadManager.py [hunk 2]
```diff
@@ -407,4 +410,28 @@ def initializeFileUploadJobMetadata(self) -> Dict[str, Any]:
...
+
+        if "makerbot" in self._formats and "makerbot" in self._file_handlers and self._file_handlers["makerbot"]:
+            filename_makerbot = self._file_name + ".makerbot"
+            metadata[filename_makerbot] = {
+                "export_job_output"   : None,
+                "upload_progress"     : -1,
+                "upload_status"       : "",
+                "file_upload_response": None,
+                "file_upload_success_message": getBackwardsCompatibleMessage(
+                    text = "'{}' was uploaded to '{}'.".format(filename_makerbot, self._library_project_name),
```
[the remark this answers is not in the document]
Comment (saumyaj3) [reply plugins/DigitalLibrary/src/DFFileExportAndUploadManager.py:422]: not needed as this was done for ufp files as well.

File: plugins/DigitalLibrary/src/DFFileExportAndUploadManager.py [hunk 3]
```diff
@@ -407,4 +410,28 @@ def initializeFileUploadJobMetadata(self) -> Dict[str, Any]:
...
+                    title = "Upload successful",
+                    message_type_str = "POSITIVE",
+                    lifetime = 30,
+                ),
+                "file_upload_failed_message": getBackwardsCompatibleMessage(
+                    text = "Failed to upload the file '{}' to '{}'.".format(filename_makerbot, self._library_project_name),
```
[the remark this answers is not in the document]
Comment (saumyaj3) [reply plugins/DigitalLibrary/src/DFFileExportAndUploadManager.py:428]: ce132e4c3f5292358a15c78ccd29aaace3e31f06

**Review by rburema — APPROVED** [review]

**Pull request merged**